pkg/apis/core/types.go

```diff
@@ -3000,12 +3017,68 @@ type PodSpec struct {
 	// - spec.containers[*].securityContext.runAsGroup
 	// +optional
 	OS *PodOS
+
 	// SchedulingGates is an opaque list of values that if specified will block scheduling the pod.
 	// More info:  https://git.k8s.io/enhancements/keps/sig-scheduling/3521-pod-scheduling-readiness.
 	//
 	// This is an alpha-level feature enabled by PodSchedulingReadiness feature gate.
 	// +optional
 	SchedulingGates []PodSchedulingGate
+
+	// ResourceClaims defines which ResourceClaims must be allocated
+	// and reserved before the Pod is allowed to start. The resources
+	// will be made available to those containers which reference them
+	// by name.
+	//
+	// This is an alpha field and requires enabling the
+	// DynamicResourceAllocation feature gate.
+	//
+	// It is immutable.
+	ResourceClaims []PodResourceClaim
+}
+
+// PodResourceClaim references exactly one ResourceClaim through a ClaimSource
+// and adds a name to it that is used inside a Pod.
+type PodResourceClaim struct {
+	// A name under which this resource can be referenced by the containers.
+	Name string
+
+	// Source determines where to find the ResourceClaim.
+	Source ClaimSource
+}
+
+// ClaimSource references exactly one ResourceClaim, either directly by name or
+// through a template for a ResourceClaim that will get created by the
+// control plane.
+//
+// Additional options might get added in the future, so code using this struct
+// must error out when none of the options that it supports are set.
+type ClaimSource struct {
+	// The resource is independent of the Pod and defined by
+	// a separate ResourceClaim in the same namespace as
+	// the Pod. Either this or Template must be set, but not both.
+	ResourceClaimName *string
+
+	// The name of a dra.k8s.io/ResourceClaimTemplate object in the same namespace.
```

Comment:
"dra" is out-dated, we went for the more generic "resource.k8s.io" based on @alculquicondor's suggestion.

Let's leave out the apigroup. I don't think it would be linked properly and users of this field need additional documentation to piece it all together anyway.